airbyte-cdk/python/airbyte_cdk/test/http/response_builder.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import functools
+import json
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from airbyte_cdk.test.http import HttpResponse
+
+
+def _extract(records_field: List[str], response_template: Dict[str, Any]) -> Any:
+    return functools.reduce(lambda a, b: a[b], records_field, response_template)
+
+
+def _replace_value(dictionary: Dict[str, Any], path: List[str], value: Any) -> None:
+    current = dictionary
+    for key in path[:-1]:
+        current = current[key]
+    current[path[-1]] = value
+
+
+class RecordBuilder:
+    def __init__(self, template: Dict[str, Any], id_path: Optional[List[str]], cursor_path: Optional[List[str]]):
+        self._record = template
+        self._id_path = id_path
+        self._cursor_path = cursor_path
+
+        self._validate_template()
+
+    def _validate_template(self) -> None:
+        paths_to_validate = [
+            ("_id_path", self._id_path),
+            ("_cursor_path", self._cursor_path),
+        ]
+        for field_name, field_path in paths_to_validate:
+            try:
+                self._validate_field(field_name, field_path)
+            except KeyError:
+                raise ValueError(f"{field_name} `{field_path}` was provided but it is not part of the template")
+
+    def _validate_field(self, field_name: str, field_path: Optional[List[str]]) -> None:
+        if field_path and not _extract(field_path, self._record):
+            raise ValueError(f"{field_name} `{field_path}` was provided but it is not part of the template")
+
+    def with_id(self, identifier: Any) -> "RecordBuilder":
+        self._set_field("id", self._id_path, identifier)
+        return self
+
+    def with_cursor(self, cursor_value: Any) -> "RecordBuilder":
+        self._set_field("cursor", self._cursor_path, cursor_value)
+        return self
+
+    def _set_field(self, field_name: str, path: Optional[List[str]], value: Any) -> None:
+        if not path:
+            raise ValueError(
+                f"{field_name}_path was not provided and hence, the record {field_name} can't be modified. Please provide `id_field` while "
+                f"instantiating RecordBuilder to leverage this capability"
+            )
+        _replace_value(self._record, path, value)
+
+    def build(self) -> Dict[str, Any]:
+        return self._record
+
+
+class HttpResponseBuilder:
+    def __init__(self, template: Dict[str, Any], records_path: List[str], pagination_strategy: Optional[Callable[[Dict[str, Any]], None]]):
```

Comment:
is it worth defining a small interface to replace the `Callable` since this will be used frequently?

Reply:
I think it does. I'll update it